### PR TITLE
Use consistent selectors across the project - Closes #3438

### LINKED ISSUE #3438: Use consistent selectors across the project
### Expected behavior
The code to select some data from the redux store is the same all across the repo but is being repeated making it harder to maintain and refactor. We should make use of selectors to access data from the redux store.

### PR DESCRIPTION
**Pull request by UsamaHameed**
### What was the problem?
This PR resolves #3438

### How was it solved?
Create selectors and custom hooks for frequently referenced information on the Redux store.

### How was it tested?
NA

### CODE REVIEW

File: src/components/screens/wallet/overview/index.js [hunk 1]
```diff
@@ -3,32 +3,23 @@ import { useSelector } from 'react-redux';
...
-};
-
 const Overview = ({
   t, activeToken, transactions, hwInfo,
   discreetMode, isWalletRoute, account,
 }) => {
-  const address = getProp(account, 'address', '');
-  const publicKey = getProp(account, 'publicKey', '');
-  const balance = getProp(account, 'balance', 0);
+  const { address, publicKey, balance } = account.info.LSK.summary;
```
Comment (reyraa) [inline src/components/screens/wallet/overview/index.js:17]: Do we need a null check here?

File: src/components/shared/transactionPriority/reducer.js [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+import { tokenMap, minAccountBalance } from '@constants';
+import { toRawLsk } from '@utils/lsk';
+
+const calculateAvailableBalance = (balance, token) => {
+  if (token !== tokenMap.LSK.key) {
+    return balance;
+  }
+  if (balance <= minAccountBalance) {
+    return balance;
+  }
```
Comment (reyraa) [inline src/components/shared/transactionPriority/reducer.js:11]: Can a user transfer this? It just seems somewhat inconstant with the next line.

Comment (reyraa) [reply src/components/shared/transactionPriority/reducer.js:11]: `Math.max(balance - minAccountBalance, 0)`

**Review by reyraa — APPROVED** [review]

File: src/components/screens/wallet/overview/index.js (hunk 1 above)
Comment (isalga) [reply src/components/screens/wallet/overview/index.js:17]: Yes, and it should be `account.summary?`

**Review by isalga — CHANGES_REQUESTED** [review]

**Review by isalga — APPROVED** [review]

**Pull request merged**